packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit/web_kit.dart

```diff
@@ -0,0 +1,71 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// The media types that require a user gesture to begin playing.
+///
+/// Wraps
+/// [WKAudiovisualMediaTypes](https://developer.apple.com/documentation/webkit/wkaudiovisualmediatypes?language=objc).
+enum AudiovisualMediaType {
+  /// No media types require a user gesture to begin playing.
+  ///
+  /// See https://developer.apple.com/documentation/webkit/wkaudiovisualmediatypes/wkaudiovisualmediatypenone?language=objc.
+  none,
+
+  /// Media types that contain audio require a user gesture to begin playing.
+  ///
+  /// See https://developer.apple.com/documentation/webkit/wkaudiovisualmediatypes/wkaudiovisualmediatypeaudio?language=objc.
+  audio,
+
+  /// Media types that contain video require a user gesture to begin playing.
+  ///
+  /// See https://developer.apple.com/documentation/webkit/wkaudiovisualmediatypes/wkaudiovisualmediatypevideo?language=objc.
+  video,
+
+  /// All media types require a user gesture to begin playing.
+  ///
+  /// See https://developer.apple.com/documentation/webkit/wkaudiovisualmediatypes/wkaudiovisualmediatypeall?language=objc.
+  all,
+}
+
+/// A collection of properties that you use to initialize a web view.
+///
+/// Wraps [WKWebViewConfiguration](https://developer.apple.com/documentation/webkit/wkwebviewconfiguration?language=objc)
+class WebViewConfiguration {
+  /// Indicates whether HTML5 videos play inline or use the native full-screen controller.
+  set allowsInlineMediaPlayback(bool allow) {
+    throw UnimplementedError();
+  }
+
+  /// The media types that require a user gesture to begin playing.
+  ///
+  /// Use [AudiovisualMediaType.none] to indicate that no user gestures are
+  /// required to begin playing media.
+  set mediaTypesRequiringUserActionForPlayback(
+    Set<AudiovisualMediaType> types,
+  ) {
+    assert(types.isNotEmpty);
+    throw UnimplementedError();
+  }
+}
+
+/// Object that displays interactive web content, such as for an in-app browser.
+///
+/// Wraps
+/// [WKWebView](https://developer.apple.com/documentation/webkit/wkwebview?language=objc).
+class WebView {
```

Comment:
I wonder if we should actually keep the WK prefix on these? It's not standard Dart, but it would make it very clear that these are just mirroring the corresponding ObjC class.

I could go either way on it, but I thought it was worth raising.

Reply:
I like this idea. I added `WK` to all the webkit classes.